### DISCUSSION TITLE
Push Notification Clarification

### DISCUSSION THREAD

**Discussion by bibinkrishnan** [GetStream/stream-chat-android#2763] [Q&A] [answered]
Hi,

I don't want the client to be connected throughout the app. I only want the client to be connected if I navigate to the chat screen. So I disconnect the user if he is not in the chat screen by 'client.disconnect()' method. . But after disconnection onMessageReceived() is getting called but unable to get sender name and message details on push notification. I tried 'disconnectsocket' it's working for me but my doubt is that after calling 'disconnectsocket' wheather it will connect the user to the chat app?. Is there any solution to get sender name and message details on push notification message?

**Comment by filbabic** [comment, marked as answer]
Hey @bibinkrishnan Sorry for the late response!

You can just choose not to connect the user until you open a given Chat. You can also disconnect the user from the socket but connect it again once you're ready to join the chat screen.
